### PR TITLE
Updated nncf_config with command line arguments

### PR DESCRIPTION
**Pull request by l-bat**

### CODE REVIEW

File: examples/tensorflow/common/sample_config.py [hunk 1]
```diff
@@ -106,6 +106,11 @@ def create_sample_config(args, parser) -> SampleConfig:
 
     sample_config = SampleConfig.from_json(args.config)
     sample_config.update_from_args(args, parser)
+
+    for key, val in sample_config.items():
+        if key in nncf_config:
```
Comment (alexsu52) [inline examples/tensorflow/common/sample_config.py:111]: If batch_size is not in nncf_config you should add it into the nncf_config too.

**Review by alexsu52 — CHANGES_REQUESTED** [review]

**Comment by vshampor** [comment]
Why would we want to mix the sample configuration and the NNCF configuration again on a same hierarchy level? I separated the `SampleConfig` from `NNCFConfig` for this exact reason. Better to make "nncf_config" a separate field in all sample .json files.

**Comment by l-bat** [comment]
> Why would we want to mix the sample configuration and the NNCF configuration again on a same hierarchy level? I separated the `SampleConfig` from `NNCFConfig` for this exact reason. Better to make "nncf_config" a separate field in all sample .json files.

`SampleConfig` can not be used because it does not pass validation using json schema. We can dump `NNCFConfig` during training and load it for testing

**Comment by alexsu52** [comment]
> 
> 
> > Why would we want to mix the sample configuration and the NNCF configuration again on a same hierarchy level? I separated the `SampleConfig` from `NNCFConfig` for this exact reason. Better to make "nncf_config" a separate field in all sample .json files.
> 
> `SampleConfig` can not be used because it does not pass validation using json schema. We can dump `NNCFConfig` during training and load it for testing

@

**Review by alexsu52 — APPROVED** [review]

**Review by vshampor — APPROVED** [review]

**Pull request merged**